### PR TITLE
Samplegen: Java sample enhancements

### LINKED ISSUE #2554: SampleGen: Java standalone samples need enhancement
- [ ] `ApiFuture` should be imported from `com.google.api.core.ApiFuture`
- [ ] Relocate the open region tag to right below the import section to make formatter happy
- [ ] Add comments for required imports for running samples
- [ ] `compile project` in gradle.build is missing a `:`

### PR DESCRIPTION
**Pull request by hzyi-google**
Fix #2554

### CODE REVIEW

**Comment by hzyi-google** [comment]
@vchudnov-g PTAL

File: src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline [hunk 1]
```diff
@@ -138,6 +137,20 @@ import java.nio.Files;
 import java.nio.Path;
 import java.nio.Paths;
 
+/*
```
Comment (vchudnov-g) [inline src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline]: I think you probably want these comments inside the region tag, don't you? The folks on the website can only see what's inside the region tags, so they won't see these comments where they are now.

And people  who can see the comments are obviously looking at this file itself, so they can see the real imports above.

File: src/main/resources/com/google/api/codegen/java/standalone_sample.snip [hunk 2]
```diff
@@ -15,11 +15,14 @@
       import org.apache.commons.cli.Option;
       import org.apache.commons.cli.Options;
     @end
-    // [START {@sample.regionTag}]
     @if sample.sampleImports.appImports.size
       {@importList(sample.sampleImports.appImports)}
 
     @end
+    @if sample.sampleImports.appImports.size
```
Comment (vchudnov-g) [inline src/main/resources/com/google/api/codegen/java/standalone_sample.snip]: Don't you want this inside the region tag? See comment in baseline.

Comment (hzyi-google) [reply src/main/resources/com/google/api/codegen/java/standalone_sample.snip]: My bad! Will fix it soon.

**Comment by hzyi-google** [comment]
PTAL @vchudnov-g

File: src/main/resources/com/google/api/codegen/java/common.snip [hunk 3]
```diff
@@ -35,6 +35,27 @@
   @end
 @end
 
+@snippet commentedImportList(fileImports)
+  /*
+   * Please include the following imports to run this sample.
```
Comment (vchudnov-g) [inline src/main/resources/com/google/api/codegen/java/common.snip]: Nit: Consider tweaking this to: "To run this sample, please include the following imports:" But I defer to @beccasaurus . It's fine as is, too.

**Comment by hzyi-google** [comment]
I'll merge it now to catch an artman release. We can adjust the comments in future PRs.

**Pull request merged**